File: src/main/java/ru/mm/surv/capture/service/FfmpegStream.java
package ru.mm.surv.capture.service;

import java.io.File;
import java.util.Collection;
import java.util.Optional;

public interface FfmpegStream {

    void start();

    void stop();

    boolean isActive();

    Collection<String> getStreamNames();

    Optional<File> getThumb(String stream);
}

File: backend/src/main/kotlin/ru/mm/surv/capture/service/FfmpegStream.kt
package ru.mm.surv.capture.service

import ru.mm.surv.dto.StreamStatus
import java.io.File

interface FfmpegStream {
    fun start()
    fun stop()
    fun status(): StreamStatus
    fun streamNames(): Collection<String>
    fun getThumb(stream: String): File?
}
